test/sequential/rolldown.withdraw.fees.test.ts

```diff
@@ -0,0 +1,164 @@
+let chain: any;
+
+beforeAll(async () => {
+  try {
+    getApi();
+  } catch (e) {
+    await initApi();
+  }
+  await setupApi();
+  api = getApi();
+  gaspIdL1Asset = JSON.parse(
+    JSON.stringify(await api.query.assetRegistry.idToL1Asset(GASP_ASSET_ID)),
+  );
+  ethIdL1Asset = JSON.parse(
+    JSON.stringify(await api.query.assetRegistry.idToL1Asset(ETH_ASSET_ID)),
+  );
+  withdrawalAmount = BN_MILLION;
+  treasuryAccount = "0x6d6f646c70792f74727372790000000000000000";
+  DEFAULT_AMOUNT = BN_THOUSAND.mul(BN_TEN.pow(new BN(18)));
+  chain = "Ethereum";
+});
+
+beforeEach(async () => {
+  [testUser] = setupUsers();
+});
+
+test("GIVEN a withdrawal, WHEN paying with GASP and withdrawing GASP, some fees goes to treasury", async () => {
+  const treasuryBalanceBefore = await api.query.tokens.accounts(
+    treasuryAccount,
+    GASP_ASSET_ID,
+  );
+  await Sudo.batchAsSudoFinalized(Assets.mintNative(testUser));
+  const events = await signTx(
+    getApi(),
+    await Withdraw(testUser, withdrawalAmount, gaspIdL1Asset.ethereum, chain),
+    testUser.keyRingPair,
+  );
+  const eventFiltered = filterZeroEventData(events, "Transfer");
+  const transferAmountBefore = eventFiltered.amount.replaceAll(",", "");
+  const transferAmount = new BN(transferAmountBefore);
+  expect(eventFiltered.from).toEqual(testUser.keyRingPair.address);
+  expect(eventFiltered.to).toEqual(treasuryAccount);
+  const treasuryBalanceAfter = await api.query.tokens.accounts(
+    treasuryAccount,
+    GASP_ASSET_ID,
+  );
+  expect(treasuryBalanceAfter.free).bnEqual(
+    treasuryBalanceBefore.free.add(transferAmount),
+  );
+});
+
+test("GIVEN a withdrawal, WHEN paying with GASP and withdrawing Eth, some fees goes to treasury", async () => {
+  const treasuryBalanceBefore = await api.query.tokens.accounts(
+    treasuryAccount,
+    GASP_ASSET_ID,
+  );
+  await Sudo.batchAsSudoFinalized(
+    Assets.mintNative(testUser),
+    Assets.mintToken(ETH_ASSET_ID, testUser, DEFAULT_AMOUNT),
+  );
+  const events = await signTx(
+    getApi(),
+    await Withdraw(testUser, withdrawalAmount, ethIdL1Asset.ethereum, chain),
+    testUser.keyRingPair,
+  );
+  const eventFiltered = filterZeroEventData(events, "Transfer");
+  const transferAmountBefore = eventFiltered.amount.replaceAll(",", "");
+  const transferAmount = new BN(transferAmountBefore);
+  expect(eventFiltered.from).toEqual(testUser.keyRingPair.address);
+  expect(eventFiltered.to).toEqual(treasuryAccount);
+  const treasuryBalanceAfter = await api.query.tokens.accounts(
+    treasuryAccount,
+    GASP_ASSET_ID,
+  );
+  expect(treasuryBalanceAfter.free).bnEqual(
+    treasuryBalanceBefore.free.add(transferAmount),
+  );
+});
+
+test("GIVEN a withdrawal, WHEN paying with GASP and withdrawing ALL GASP, extrinsic fail", async () => {
+  await Sudo.batchAsSudoFinalized(Assets.mintNative(testUser));
+  const tokenAmount = await api.query.tokens.accounts(
+    testUser.keyRingPair.address,
+    GASP_ASSET_ID,
+  );
```

Comment:
is not there any getter?  If there is not, plz create one in "Asset" static class,

Reply:
I found, it was changed